### DISCUSSION TITLE
Unable to determine underlying reason for email alert send failure

### DISCUSSION THREAD

**Discussion by kolban-google** [apache/airflow#18081] [General] [answered]
### Apache Airflow version

2.1.2

### Operating System

Debian

### Versions of Apache Airflow Providers

Google Composer

### Deployment

Composer

### What happened

I was trying to configure sending of alerts via SMTP.  I must have made a configuration error and I get the error "Failed to send".  Searching in the code I find the following:

https://github.com/apache/airflow/blob/944dcfbb918050274fd3a1cc51d8fdf460ea2429/airflow/models/taskinstance.py#L1719

What this shows is:

```
if email_for_state and task.email:
            try:
                self.email_alert(error)
            except Exception:
                self.log.exception('Failed to send email to: %s', task.email)
```

what this means is that if an exception occurs while sending an email, all we get is a "Failed to send" error and the diagnostics of why we failed to send have been lost.

### What you expected to happen

I would like to see additional debuging on why the email failed to send.  Presumably this would be the exception text from the underlying call to send an email.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment, marked as answer]
I believe this is invalid. ``self.log.exception`` also shows exception information and you should see it in the log in this case https://docs.python.org/3/library/logging.html#logging.Logger.exception 

```
try:
    raise Exception("exc message")
except Exception:
    log.exception("test")
test
Traceback (most recent call last):
  File "<ipython-input-5-949bdc201447>", line 2, in <module>
    raise Exception("exc message")
Exception: exc message
```

**Comment by potiuk** [comment]
I will close it fro now, please pas some logs if you think the message is unclear.

But If you would like to improve it and add some more information, I will invite you to open a PR directly. I will convert it into discussion now if you want to provide more information and discuss better approaches.

**Reply by kolban-google** [reply]
I just wanted to say thank you.  After your post I went back and looked really carefully at the logs and ... sure enough ... the exceptions I had hoped for had been there all the time.  Your response was fantastic and ultra quick.  THANK YOU my friend.
